src/core.js

```diff
@@ -0,0 +1,76 @@
+import { unlerp } from '@kaliber/math'
+
+function calculateScrollProgression({ element, container, start, end }) {
+  const scrollPosition = container === window ? window.scrollY : container.scrollTop
+  return unlerp({
+    start: calculatePosition({ element, container, position: start, scrollPosition }),
+    end: calculatePosition({ element, container, position: end, scrollPosition }),
+    clamp: true,
+    input: scrollPosition
+  })
+}
+
+function calculatePosition({ element, container, position, scrollPosition }) {
+  const elementRect = element.getBoundingClientRect()
+  const containerRect = container === window ? getWindowRect() : container.getBoundingClientRect()
+
+  const offsetTop = elementRect.top + scrollPosition - containerRect.top
+  const elementPosition = position.element.anchor * elementRect.height + (position.element.offset ?? 0)
+  const containerPosition = position.container.anchor * containerRect.height + (position.container.offset ?? 0)
+
+  return offsetTop + elementPosition - containerPosition
+}
+
+export function onScrollProgression({ element, start, end, onChange }) {
+  const container = getOverflowContainer(element) ?? window
```

Comment:
Het is wel slim om dit in de documentatie op te nemen, dat by default de 'overflow' container met waarde `auto` of `scroll` wordt gepakt als container.